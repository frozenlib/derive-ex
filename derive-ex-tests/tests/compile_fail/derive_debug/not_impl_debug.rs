fn main() {}

struct NotImplDebug;
#[derive(derive_ex::Ex)]
#[derive_ex(Debug)]
struct X {
    field: NotImplDebug,
}
