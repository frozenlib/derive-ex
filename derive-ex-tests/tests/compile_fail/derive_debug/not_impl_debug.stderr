error[E0277]: `NotImplDebug` doesn't implement `Debug`
 --> tests/compile_fail/derive_debug/not_impl_debug.rs:4:10
  |
4 | #[derive(derive_ex::Ex)]
  |          ^^^^^^^^^^^^^ the trait `Debug` is not implemented for `NotImplDebug`
  |
  = note: add `#[derive(Debug)]` to `NotImplDebug` or manually `impl Debug for NotImplDebug`
  = note: required for the cast from `&NotImplDebug` to `&dyn Debug`
  = note: this error originates in the derive macro `derive_ex::Ex` (in Nightly builds, run with -Z macro-backtrace for more info)
help: consider annotating `NotImplDebug` with `#[derive(Debug)]`
  |
3 + #[derive(Debug)]
4 | struct NotImplDebug;
  |
